src/hooks/useCanClaimReward.ts

```diff
@@ -0,0 +1,52 @@
+import { useGetExternalEligibility } from './useGetExternalEligibility'
+import { canClaimReward } from '@/helpers/rewards'
+import { useGetQuest } from './useGetQuest'
+import { useGetUserPlayStreak } from './useGetUserPlayStreak'
+import { useQuestWrapper } from '@/state/QuestWrapperProvider'
+
+export function useCanClaimReward({ questId }: { questId: number }) {
+  const { getQuest, getExternalEligibility, getUserPlayStreak } =
+    useQuestWrapper()
+
+  const { data: questQuery, isLoading: isQuestLoading } = useGetQuest(
+    questId,
+    getQuest
+  )
+
+  const { data: externalEligibility, isLoading: isExternalEligibilityLoading } =
+    useGetExternalEligibility({
+      questId,
+      getExternalEligibility: getExternalEligibility
+    })
+
+  const { data: playstreakQuery, isLoading: isUserPlayStreakLoading } =
+    useGetUserPlayStreak(questId, getUserPlayStreak)
+
+  const isLoading =
+    isQuestLoading || isExternalEligibilityLoading || isUserPlayStreakLoading
+
+  if (isLoading) {
+    return {
+      canClaim: false,
+      isLoading: true
+    }
+  }
+
+  if (!questQuery.data) {
+    return {
+      canClaim: false,
+      isLoading: false
+    }
+  }
+
+  const canClaim = canClaimReward({
+    quest: questQuery.data,
+    externalEligibility,
+    playstreakData: playstreakQuery.data?.userPlayStreak
+  })
+
```

Comment:
this is a bit of an anti-pattern. can we use the TS functions in a single react query `useQuery` here? That way we only fetch the quest eligibility that matches the type, we have 1 hook with simplified fetching, loading, etc. semantics and we can more easily support compound eligibility types in the future?

Reply:
I think this is a good idea so I added it, lmk what you think

Reply:
yeah. perfect